File: app/src/main/java/com/texnar13/deliveryapp/ui/TrajectoriesFragment.java
package com.texnar13.deliveryapp.ui;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import com.texnar13.deliveryapp.R;
import com.texnar13.deliveryapp.view_model.MainViewModel;

public class TrajectoriesFragment extends Fragment {


    // Required empty public constructor
    public TrajectoriesFragment() {}


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_trajectories, container, false);

        MainViewModel mainViewModel = new ViewModelProvider(requireActivity()).get(MainViewModel.class);

// ------------------------------------------- разметка -------------------------------------------

        // контейнер trip
        LinearLayout boxesContainer = rootView.findViewById(R.id.fragment_trajectories_search_result_container);

// -------------------------------- подписываемся на изменения во viewModel --------------------------------

//        // глобальное изменение переменной
//        mainViewModel.currentLoadedTrips.observe(this, dbUsers -> {
//
//            // вывод списка
//            boxesContainer.removeAllViews();

//            // проходимся по всем отправлениям
//            for (DBTrip tripUnit : dbUsers) {
//
//                // создание разметки
//                View notificationViewElement = getLayoutInflater().inflate(R.layout.element_trip_box, null);
//
//
//                TextView username = notificationViewElement.findViewById(R.id.element_trip_box_username);
//                TextView description = notificationViewElement.findViewById(R.id.element_trip_box_description);
//                TextView joinButton = notificationViewElement.findViewById(R.id.element_trip_box_join_button);
//
//
//
//                //username.setText(tripUnit.get().getName());
//
//
//
//
//                description.setText(String.format(
//                        Locale.getDefault(), "Категория %s \nиз %s, %s -> в %s, %s\n" +
//                                "Описание %s \nВес %.1fКГ\nГабариты %.1fx%.1fx%.1fсм",
//                        expeditionUnit.getPackage().getCategory(),
//                        expeditionUnit.getAddressSender().getArray()[0],
//                        expeditionUnit.getAddressSender().getArray()[1],
//                        expeditionUnit.getAddressReceiver().getArray()[0],
//                        expeditionUnit.getAddressReceiver().getArray()[1],
//                        expeditionUnit.getPackage().getDescription(),
//                        expeditionUnit.getPackage().getWeight(),
//                        expeditionUnit.getPackage().getDimensions()[0],
//                        expeditionUnit.getPackage().getDimensions()[1],
//                        expeditionUnit.getPackage().getDimensions()[2]
//                ));
//
//                // статус
//                switch (expeditionUnit.getStatus()) {
//                    case DBExpedition.EXPEDITION_STATUS_VALUE_WAIT_SEND:
//                        // выставляем статус
//                        state.setText("Ожидает отправки");
//                        state.setTextColor(getResources().getColor(R.color.wait_mail_text_color));
//
//                        // кнопки
//                        buttonEdit.setOnClickListener(v -> {
//                            // вызов диалога
//                            ExpeditionEditDialogFragment.newInstance(expeditionUnit)
//                                    .show(getParentFragmentManager(), "ExpeditionEditDialogFragment");
//                        });
//                        buttonFindSomebody.setOnClickListener(v -> {
//                            state.setText("buttonFindSomebody");
//                        });
//                        buttonConnect.setVisibility(View.INVISIBLE);
//                        break;
//                    case DBExpedition.EXPEDITION_STATUS_VALUE_SENT:
//                        // выставляем статус
//                        state.setText("ОТПРАВЛЕНО");
//                        state.setTextColor(getResources().getColor(R.color.sent_mail_text_color));
//
//                        // кнопки
//                        buttonEdit.setOnClickListener(v -> {
//                            // вызов диалога
//                            ExpeditionEditDialogFragment.newInstance(expeditionUnit)
//                                    .show(getParentFragmentManager(), "ExpeditionEditDialogFragment");
//                        });
//                        buttonFindSomebody.setVisibility(View.INVISIBLE);
//                        buttonConnect.setOnClickListener(v -> {
//                            state.setText("buttonConnect");
//                        });
//                        break;
//                    case DBExpedition.EXPEDITION_STATUS_VALUE_DONE:
//                        // выставляем статус
//                        state.setText("Завершено");
//                        state.setTextColor(getResources().getColor(R.color.ended_mail_text_color));
//
//                        // кнопки
//                        buttonEdit.setVisibility(View.INVISIBLE);
//                        buttonFindSomebody.setVisibility(View.INVISIBLE);
//                        buttonConnect.setVisibility(View.INVISIBLE);
//                        break;
//                }
//
//
//                LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(
//                        LinearLayout.LayoutParams.MATCH_PARENT,
//                        LinearLayout.LayoutParams.WRAP_CONTENT
//                );
//                layoutParams.topMargin = getResources().getDimensionPixelOffset(R.dimen.containers_margin);
//
//                boxesContainer.addView(
//                        notificationViewElement,layoutParams
//                );
//            }
//        });



        return rootView;
    }
}

File: app/src/main/java/com/texnar13/deliveryapp/ui/TrajectoriesFragment.kt
package com.texnar13.deliveryapp.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.google.android.material.textfield.TextInputLayout
import com.texnar13.deliveryapp.R
import com.texnar13.deliveryapp.model.entities.EntityTrip
import com.texnar13.deliveryapp.view_model.MainViewModel
import java.lang.String
import java.util.Locale

class TrajectoriesFragment  // Required empty public constructor
    : Fragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment

        val rootView = inflater.inflate(R.layout.fragment_trajectories, container, false)

        // ------------------------------------------- разметка -------------------------------------------

        val textViewDepartCountry = rootView.findViewById<TextInputLayout>(R.id.fragment_trajectories_search_depart_country).editText!!
        val textViewDepartCity = rootView.findViewById<TextInputLayout>(R.id.fragment_trajectories_search_depart_city).editText!!
        val textViewArriveCountry = rootView.findViewById<TextInputLayout>(R.id.fragment_trajectories_search_arrive_country).editText!!
        val textViewArriveCity = rootView.findViewById<TextInputLayout>(R.id.fragment_trajectories_search_arrive_city).editText!!
        val textViewDepartDate = rootView.findViewById<TextInputLayout>(R.id.fragment_trajectories_search_depart_date).editText!!
        val textViewFreeWeight = rootView.findViewById<TextInputLayout>(R.id.fragment_trajectories_search_free_weight).editText!!
        val searchButton = rootView.findViewById<Button>(R.id.fragment_trajectories_search_button)








        // контейнер trip
        val boxesContainer = rootView.findViewById<LinearLayout>(R.id.fragment_trajectories_search_result_container)

        // -------------------------------- подписываемся на изменения во viewModel --------------------------------
        val mainViewModel = MainViewModel.getViewModel(requireActivity())

        // смотрим за отправлением, если оно есть вводим его параметры в поиск маршрутов
        mainViewModel.selectedExpedition.observe(viewLifecycleOwner) {
            if (it != null) {
                // Выставляем данные в поля

                textViewDepartCountry.setText(it.addressSender.address[0])
                textViewDepartCity.setText(it.addressSender.address[1])
                textViewArriveCountry.setText(it.addressReceiver.address[0])
                textViewArriveCity.setText(it.addressReceiver.address[1])
                textViewDepartDate.setText("")
                textViewFreeWeight.setText(it.expeditionPackage.weight.toString())
            }
        }

        mainViewModel.loadTrips()


        // глобальное изменение переменной
        mainViewModel.currentLoadedTrips.observe(viewLifecycleOwner) { trips ->

            // вывод списка
            outList(trips, boxesContainer)
        }


        // кнопка поиска
        searchButton.setOnClickListener {
            mainViewModel.loadTrips()
        }

        return rootView
    }


    // вывод списка
    private fun outList(trips: List<EntityTrip>, outputContainer: LinearLayout){
        outputContainer.removeAllViews()

        // проходимся по всем отправлениям
        for (tripUnit in trips) {
            // создание разметки

            val notificationViewElement = layoutInflater.inflate(R.layout.element_trip_box, null)


            val username: TextView = notificationViewElement.findViewById(R.id.element_trip_box_username)
            val description: TextView = notificationViewElement.findViewById(R.id.element_trip_box_description)
            val joinButton: TextView = notificationViewElement.findViewById(R.id.element_trip_box_join_button)


            description.text = String.format(
                    Locale.getDefault(), """
                            Отправляюсь %s
                            из %s, %s -> в %s, %s
                            Могу взять %fКг веса
                            """.trimIndent(),
                    tripUnit.sentDate.toString(),
                    tripUnit.sendCountry,
                    tripUnit.sendCity,
                    tripUnit.receivingCountry,
                    tripUnit.receivingCity,
                    tripUnit.availableWeight
            )


            val layoutParams = LinearLayout.LayoutParams(
                    LinearLayout.LayoutParams.MATCH_PARENT,
                    LinearLayout.LayoutParams.WRAP_CONTENT
            )
            layoutParams.topMargin = resources.getDimensionPixelOffset(R.dimen.containers_margin)

            outputContainer.addView(
                    notificationViewElement, layoutParams
            )
        }
    }


}